### PR TITLE
Granular permissions^2

### LINKED ISSUE #488: New update requires permission for all website data?
Prior to this most recent update from what I could tell, the only data needed was for permission for the relevant websites this addon deals with.

Why does it suddenly need access to all my data for web sites?

### PR DESCRIPTION
**Pull request by TheGoddessInari**
Because I'm being nice today even though
>  iamadamdev commented 16 hours ago
> Cause I got lazy.

Fixes #488.

### CODE REVIEW

**Comment by joker327** [comment]
You removed 'general paywall bypass' though.

**Comment by TheGoddessInari** [comment]
It's not going to be very useful on the already covered sites, is it?

**Comment by joker327** [comment]
Unless it's intended as backup mechanism and for new (not yet covered) sites.

**Comment by TheGoddessInari** [comment]
Which wouldn't occur with granular permissions. :)

**Comment by joker327** [comment]
BP depended on uBlock Origin for general paywall bypass (also for partly covered sites). This was a (crude) way to fix this, I guess.
PS paywall sites could also be added to permissions though.

**Comment by TheGoddessInari** [comment]
I'm not sure what you mean. Granular permissions was a PR I did quite some time ago because @iamadamdev said they were too lazy to do it. So up until 22 hours ago, that's the way it was working.
Now @iamadamdev says they're too lazy again so removed it, even though `defaultSites` was duplicated, and ~100% of the code (aside from `manifest.json` and Google Analytics in the chrome version) is shared, just with some tweaks needed.

See #488. Nobody wants an addon that demands wide permissions when completely unnecessary.

I was just going to hard fork because
> iamadamdev commented 16 hours ago
> Cause I got lazy.

but I decided to be nice and make another PR instead.

**Comment by joker327** [comment]
To block external paywall scripts, you have to add their domains to the permissions.
It's necessary to bypass some 'covered' sites.

File: background.js [hunk 1]
```diff
@@ -291,15 +189,13 @@ browser.runtime.onInstalled.addListener(function(details) {
...
-    return enabledSite.indexOf("generalpaywallbypass") !== -1
-  })) {
+  if (!isSiteEnabled(details) || details.url.indexOf("mod=rsswn") !== -1) {
     return;
   }
-  return {cancel: true}; 
+  return {cancel: true};
   },
   {
-    urls: ["*://*.newstatesman.com/*", "*://*.tinypass.com/*", "*://*.poool.fr/*", "*://*.piano.io/*", "*://*.outbrain.com/*"],
```
Comment (iamadamdev) [inline background.js]: Can you leave these? It is for the general bypass

File: README.md [hunk 2]
```diff
@@ -125,7 +125,6 @@ Usually premium articles cannot be bypassed as they are behind a hard paywall.
 ### Troubleshooting
 * This add-on works best alongside [uBlock Origin](https://addons.mozilla.org/en-US/firefox/addon/ublock-origin/).
 * If a site doesn't work try turning off uBlock and refreshing.
-* If a site is having problems try unchecking "\*General Paywall Bypass\*" in Options.
```
Comment (iamadamdev) [inline README.md]: Please leave this to cover the non-listed sites

**Comment by Armybrat20** [comment]
Question: Is Bypass-Paywalls compatible with earlier versions of Firefox or just the most recent? I installed it on a machine here that's running Firefox 62.0.3 and although it installed with no errors, I notice that the drop-down is only 'Options' and 'Git Hub'. On my own machine, I get a long dropdown with all the sites listed and the ability to disable/enable each, individually. 

And on that 62.0.3 machine, it does not seem to be bypassing paywalls.

**Comment by iamadamdev** [comment]
@Armybrat20 The latest version is not, however at some time it probably was. You might want to go back in the releases and try a much older version.

**Pull request merged**